src/osd/ECBackend.cc

```diff
@@ -1431,9 +1431,24 @@ void ECBackend::filter_read_op(
   }
 
   if (op.in_progress.empty()) {
+    // Determine recovery cost
+    int cost = 0;
+    auto ropiter = tid_to_read_map.find(op.tid);
+    if (ropiter != tid_to_read_map.end()) {
+      ReadOp &rop = ropiter->second;
+      for (auto &&i: rop.complete) {
+        for (auto &&ret: i.second.returned) {
+          for (auto &&shard: ret.get<2>()) {
+            cost += shard.second.length();
+          }
+        }
+      }
+    }
```

Comment:
See my branch for a comment with an explanation, but I think we want a cost of 1 here.

Reply:
Thanks for the detailed explanation of the case involved.